src/python/merge_statistics.py

```diff
@@ -0,0 +1,283 @@
+    Args:
+        repo (Repository): The repository object.
+        left_sha (str): The left sha.
+        right_sha (str): The right sha.
+    Returns:
+        str: A string containing the diff result.
+    """
+    command = f"git diff {left_sha} {right_sha}"
+    stdout, _ = repo.run_command(command)
+    return stdout
+
+
+def compute_num_diff_lines(
+        repo: Repository, left_sha: str, right_sha: str
+) -> Union[int, None]:
+    """
+    Computes the number of lines that are different between two branches using git diff.
+    Args:
+        repo (Repository): The repository object.
+        left_sha (str): The left sha.
+        right_sha (str): The right sha.
+    """
+    try:
+        diff = get_diff(repo, left_sha, right_sha, None)
+    except Exception as e:
+        logger.error(
+            f"compute_num_diff_lines: {left_sha} {right_sha} {repo.repo_slug} {e}"
+        )
+        return None
+    return sum(1 for line in diff.splitlines() if line.startswith(("+ ", "- ")))
+
+
+def compute_imports_involved(
+        repo: Repository, left_sha: str, right_sha: str
+) -> Union[bool, None]:
+    """
+    Computes if imports are involved in the diff between two branches using git diff.
+    Args:
+        repo (Repository): The repository object.
+        left_sha (str): The left sha.
+        right_sha (str): The right sha.
+    Returns:
+        bool: True if imports are involved, False otherwise.
+    """
+    try:
+        diff = get_diff(repo, left_sha, right_sha, None)
+    except Exception as e:
+        logger.error(
+            f"compute_imports_involved: {left_sha} {right_sha} {repo.repo_slug} {e}"
+        )
+        return None
+    return "import " in diff
+
+
+def diff_contains_non_java_file(
+        repo: Repository, left_sha: str, right_sha: str
+) -> Union[bool, None]:
+    """
+    Computes the diff between two branches using git diff.
+    Args:
+        repo (Repository): The repository object.
+        left_sha (str): The left sha.
+        right_sha (str): The right sha.
+    Returns:
+        bool: True if the diff contains a non-java file, False otherwise.
+    """
+    try:
+        merge_diff = get_diff_files(repo, left_sha, right_sha)
+    except Exception as e:
+        logger.error(
+            f"diff_contains_non_java_file: {left_sha} {right_sha} {repo.repo_slug} {e}"
+        )
+        return None
+    return any(not file.endswith(".java") for file in merge_diff)
+
+
+def compute_statistics(
+        merge_idx: str,
+        repo_slug: str,
+        merge_data: pd.Series,
+) -> pd.DataFrame:
+    """
+    Compute statistics for a merge.
+    Args:
+        merge_idx (str): The merge index.
+        repo_slug (str): The repository slug.
+        merge_data (pd.Series): The merge data.
+    Returns:
+        pd.DataFrame: A dataframe containing the statistics.
+    """
+    # Create row results.
+    statistics = {"idx": merge_idx}
+
+    # Clone base, left, right branches.
```

Comment:
Do you need all three for your purposes?  If you are just using `git` commands, I think you can clone the repo once, because all the shas are accessible from it.

Reply:
Did not realize I didn't have to check-out each branch to use the repo object! Makes everything so much faster.